cli/operator/keys/import_test.go

```diff
@@ -0,0 +1,134 @@
+			name: "empty key name argument",
+			args: []string{"", ""},
+			err:  ErrEmptyKeyName,
+		},
+		{
+			name: "keyname with whitespaces",
+			args: []string{"hello world", ""},
+			err:  ErrKeyContainsWhitespaces,
+		},
+		{
+			name: "empty private key argument",
+			args: []string{"hello", ""},
+			err:  ErrEmptyPrivateKey,
+		},
+		{
+			name: "keyname with whitespaces",
+			args: []string{"hello", "hello world"},
+			err:  ErrPrivateKeyContainsWhitespaces,
+		},
+		{
+			name: "invalid keytype",
+			args: []string{"--key-type", "invalid", "hello", "privkey"},
+			err:  ErrInvalidKeyType,
+		},
+		{
+			name: "invalid password based on validation function - ecdsa",
+			args: []string{"--key-type", "ecdsa", "test", "6842fb8f5fa574d0482818b8a825a15c4d68f542693197f2c2497e3562f335f6"},
+			err:  ErrInvalidPassword,
+			promptMock: func(p *prompterMock.MockPrompter) {
+				p.EXPECT().InputHiddenString(gomock.Any(), gomock.Any(), gomock.Any()).Return("", ErrInvalidPassword)
+			},
+		},
+		{
+			name: "invalid password based on validation function - bls",
+			args: []string{"--key-type", "bls", "test", "123"},
+			err:  ErrInvalidPassword,
+			promptMock: func(p *prompterMock.MockPrompter) {
+				p.EXPECT().InputHiddenString(gomock.Any(), gomock.Any(), gomock.Any()).Return("", ErrInvalidPassword)
+			},
+		},
+		{
+			name: "valid ecdsa key import",
+			args: []string{"--key-type", "ecdsa", "test", "6842fb8f5fa574d0482818b8a825a15c4d68f542693197f2c2497e3562f335f6"},
+			err:  nil,
+			promptMock: func(p *prompterMock.MockPrompter) {
+				p.EXPECT().InputHiddenString(gomock.Any(), gomock.Any(), gomock.Any()).Return("", nil)
+			},
+			keyPath: OperatorKeyFolder + "/test.ecdsa.key.json",
+		},
+		{
+			name: "valid ecdsa key import with 0x prefix",
+			args: []string{"--key-type", "ecdsa", "test", "0x6842fb8f5fa574d0482818b8a825a15c4d68f542693197f2c2497e3562f335f6"},
+			err:  nil,
+			promptMock: func(p *prompterMock.MockPrompter) {
+				p.EXPECT().InputHiddenString(gomock.Any(), gomock.Any(), gomock.Any()).Return("", nil)
+			},
+			keyPath: OperatorKeyFolder + "/test.ecdsa.key.json",
+		},
+		{
+			name: "valid bls key import",
+			args: []string{"--key-type", "bls", "test", "20030410000080487431431153104351076122223465926814327806350179952713280726583"},
+			err:  nil,
+			promptMock: func(p *prompterMock.MockPrompter) {
+				p.EXPECT().InputHiddenString(gomock.Any(), gomock.Any(), gomock.Any()).Return("", nil)
+			},
+			keyPath: OperatorKeyFolder + "/test.bls.key.json",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Cleanup(func() {
+				_ = os.RemoveAll(OperatorKeyFolder)
+			})
+			controller := gomock.NewController(t)
+			p := prompterMock.NewMockPrompter(controller)
+			if tt.promptMock != nil {
+				tt.promptMock(p)
+			}
+
+			importCmd := ImportCmd(p)
+			importCmd.SetArgs(tt.args)
+			err := importCmd.Execute()
+
+			if tt.err == nil {
+				assert.NoError(t, err)
+				_, err := os.Stat(tt.keyPath)
```

Comment:
Can we assert the contents of the key file somehow? The key file might be created but it can be empty or with the wrong contents.

Reply:
Hm yea. Good catch. Let me add those.